kv: Restore SendNextTimeout

Build sha: 210f5b0e9f12660120fe6d483a8aa8640d239df4 recently upgraded from cc15235

node 0 (`104.196.97.139`) crashed with:
```
panic: runtime error: index out of range

goroutine 1166859 [running]:
panic(0x17012a0, 0xc820012030)
        /usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/roachpb.(*HeartbeatTxnRequest).MarshalTo(0xc831dbd0c0, 0xc841518110, 0x37, 0x37, 0x45, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:2714 +0x2d8
github.com/cockroachdb/cockroach/roachpb.(*RequestUnion).MarshalTo(0xc8310e5570, 0xc84151810e, 0x39, 0x39, 0x47, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:3577 +0x1004
github.com/cockroachdb/cockroach/roachpb.(*BatchRequest).MarshalTo(0xc832b30d18, 0xc841518000, 0x147, 0x147, 0x147, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:4166 +0x304
github.com/cockroachdb/cockroach/roachpb.(*BatchRequest).Marshal(0xc832b30d18, 0xc841518000, 0x147, 0x147, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:4141 +0xa0
github.com/golang/protobuf/proto.Marshal(0x7f00178d4c48, 0xc832b30d18, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/golang/protobuf/proto/encode.go:225 +0xca
google.golang.org/grpc.protoCodec.Marshal(0x18680c0, 0xc832b30d18, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/google.golang.org/grpc/rpc_util.go:68 +0x89
google.golang.org/grpc.(*protoCodec).Marshal(0x26d4bf0, 0x18680c0, 0xc832b30d18, 0x0, 0x0, 0x0, 0x0, 0x0)
        <autogenerated>:20 +0xbe
google.golang.org/grpc.encode(0x7f001784d948, 0x26d4bf0, 0x18680c0, 0xc832b30d18, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/google.golang.org/grpc/rpc_util.go:246 +0xb4
google.golang.org/grpc.sendRequest(0x7f00178d3e80, 0xc831c24510, 0x7f001784d948, 0x26d4bf0, 0x0, 0x0, 0xc831af4050, 0x7f00178847d8, 0xc8225ea780, 0x18680c0, ...)
        /go/src/google.golang.org/grpc/call.go:88 +0x1b1
google.golang.org/grpc.Invoke(0x7f00178d3e80, 0xc831c24510, 0x19b8c00, 0x21, 0x18680c0, 0xc832b30d18, 0x1813920, 0xc841807860, 0xc822f700b0, 0x0, ...)
        /go/src/google.golang.org/grpc/call.go:166 +0x943
github.com/cockroachdb/cockroach/roachpb.(*internalClient).Batch(0xc837a11290, 0x7f00178d3e80, 0xc831c24510, 0xc832b30d18, 0x0, 0x0, 0x0, 0xc831c24510, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:1409 +0xec
github.com/cockroachdb/cockroach/kv.(*grpcTransport).SendNext.func1(0xc8293770e0, 0xc832b30cf0, 0xc82299e4e0, 0xc820b22400, 0x16)
        /go/src/github.com/cockroachdb/cockroach/kv/transport.go:223 +0x372
created by github.com/cockroachdb/cockroach/kv.(*grpcTransport).SendNext
        /go/src/github.com/cockroachdb/cockroach/kv/transport.go:225 +0x4f3
```

Node log (full log for last day or so):
[node0.log.txt.gz](https://github.com/cockroachdb/cockroach/files/268553/node0.log.txt.gz)

And this one from `beta`: also marshalling within BatchRequest.
```
I160518 06:58:07.026947 storage/replica_command.go:1477  range 281: replicaID 5 transfer raft leadership to replicaID 2
panic: runtime error: index out of range

goroutine 3047465 [running]:
panic(0x17012a0, 0xc820012030)
        /usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/roachpb.(*ReverseScanRequest).MarshalTo(0xc833186300, 0xc825ed8d45, 0x22, 0x22, 0x34, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:2126 +0x1d4
github.com/cockroachdb/cockroach/roachpb.(*RequestUnion).MarshalTo(0xc827cc9570, 0xc825ed8d42, 0x25, 0x25, 0x37, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:3679 +0x1cbe
github.com/cockroachdb/cockroach/roachpb.(*BatchRequest).MarshalTo(0xc832fe64a8, 0xc825ed8c60, 0x107, 0x107, 0x107, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:4166 +0x304
github.com/cockroachdb/cockroach/roachpb.(*BatchRequest).Marshal(0xc832fe64a8, 0xc825ed8c60, 0x107, 0x107, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:4141 +0xa0
github.com/golang/protobuf/proto.Marshal(0x7fe37b649790, 0xc832fe64a8, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/golang/protobuf/proto/encode.go:225 +0xca
google.golang.org/grpc.protoCodec.Marshal(0x18680c0, 0xc832fe64a8, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/google.golang.org/grpc/rpc_util.go:68 +0x89
google.golang.org/grpc.(*protoCodec).Marshal(0x26d4bf0, 0x18680c0, 0xc832fe64a8, 0x0, 0x0, 0x0, 0x0, 0x0)
        <autogenerated>:20 +0xbe
google.golang.org/grpc.encode(0x7fe37b756be0, 0x26d4bf0, 0x18680c0, 0xc832fe64a8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/google.golang.org/grpc/rpc_util.go:246 +0xb4
google.golang.org/grpc.sendRequest(0x7fe37b70cb18, 0xc8250c6450, 0x7fe37b756be0, 0x26d4bf0, 0x0, 0x0, 0xc82351b360, 0x7fe37b769da8, 0xc822a41d10, 0x18680c0, ...)
        /go/src/google.golang.org/grpc/call.go:88 +0x1b1
google.golang.org/grpc.Invoke(0x7fe37b70cb18, 0xc8250c6450, 0x19b8c00, 0x21, 0x18680c0, 0xc832fe64a8, 0x1813920, 0xc822103320, 0xc8233e8f20, 0x0, ...)
        /go/src/google.golang.org/grpc/call.go:166 +0x943
github.com/cockroachdb/cockroach/roachpb.(*internalClient).Batch(0xc8291934a8, 0x7fe37b70cb18, 0xc8250c6450, 0xc832fe64a8, 0x0, 0x0, 0x0, 0xc8250c6450, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:1409 +0xec
github.com/cockroachdb/cockroach/kv.(*grpcTransport).SendNext.func1(0xc823ba3630, 0xc832fe6480, 0xc82cd805a0, 0xc82319cf80, 0x16)
        /go/src/github.com/cockroachdb/cockroach/kv/transport.go:223 +0x372
created by github.com/cockroachdb/cockroach/kv.(*grpcTransport).SendNext
        /go/src/github.com/cockroachdb/cockroach/kv/transport.go:225 +0x4f3
```

--- a/kv/dist_sender.go
+++ b/kv/dist_sender.go
@@ -42,10 +42,7 @@ import (
 const (
 	// TODO(bdarnell): make SendNextTimeout configurable.
 	// https://github.com/cockroachdb/cockroach/issues/6719
-	//
-	// Temporarily increased from 500ms to 10s; see
-	// https://github.com/cockroachdb/cockroach/issues/6750
-	defaultSendNextTimeout = 10 * time.Second
+	defaultSendNextTimeout = 500 * time.Millisecond
 	defaultClientTimeout   = 10 * time.Second
 
 	// The default maximum number of ranges to return from a range

